### PR TITLE
#2470: (Updated) reduce sum, max for dim=[N,C,H,W][2] will need trans…

### PR DESCRIPTION
**Pull request by muthutt**
will need transpose-compute-transpose on the WH B0 architecture today

- This reverts commit 5fc3c9a3cee0ab42fdab1e43a20d34f7a945d341.
- applying workaround strictly once is required

### CODE REVIEW

**Review by tt-aho — APPROVED** [review]

File: tests/tt_eager/ops/test_reduce_op.cpp [hunk 1]
```diff
@@ -99,9 +99,8 @@ int main () {
     auto some_tensor = tt::numpy::random::uniform(bfloat16(0.0f), bfloat16(0.0f), {1, 1, 32, 32}).to(Layout::TILE).to(device);
 
     run_reduce_ops();
-
     TT_FATAL(
-        tt::tt_metal::program_cache::num_entries() == 6,
+        tt::tt_metal::program_cache::num_entries() >= 5,
```
Comment (arakhmati) [inline tests/tt_eager/ops/test_reduce_op.cpp:103]: Why is it `>=`?

**Review by arakhmati — CHANGES_REQUESTED** [review]

**Review by TT-billteng — CHANGES_REQUESTED** [review]
UMD

**Review by TT-billteng — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Review by tt-rkim — APPROVED** [review]

File: tt_eager/tt_dnn/op_library/reduce/reduce_op.cpp [hunk 2]
```diff
@@ -210,10 +220,75 @@ Tensor mean(const Tensor& input_tensor,uint aggregate_dims /* = 2 */, const Memo
...
+}
+
 Tensor sum(const Tensor &input_tensor, uint dim, const MemoryConfig& output_mem_config) {
     TT_ASSERT( dim >= 0 && dim <= 3, "dimension have to be 0-3 only corresponding to N,C,H,W");
     constexpr float scaler1 = 1.0;
 
+    if (is_arch_whb0(input_tensor.device()->arch())) {
+      return reduce_impl<ReduceOpMath::SUM>(input_tensor, dim,output_mem_config);
+    }
+
```
Comment (tt-aho) [inline tt_eager/tt_dnn/op_library/reduce/reduce_op.cpp:291]: Why can't this just call `reduce_impl<ReduceOpMath::SUM>` like in the case for max?

What is the difference?

**Review by tt-aho — CHANGES_REQUESTED** [review]

**Comment by tt-aho** [comment]
So is the plan to move directly to an actual fix for the cause of the issue, and not merge this workaround at all?

**Pull request closed without merging**